app/views/wishes/_filter.html.erb

```diff
@@ -9,35 +9,39 @@
       <%# Sorting %>
       <div class="dropdown">
         <button class="p-0" type="button" data-bs-toggle="dropdown" aria-expanded="false">
-          <%= image_tag 'sort.png', alt: "sort-icon", class: ("active" if params[:sort_by]) %>
+          <%= image_tag 'sort.png', alt: "sort-icon", class: ("active" if params[:order_column]) %>
         </button>
         <ul class="dropdown-menu">
           <li class="dropdown-header">SORT BY</li>
           <li><hr class="dropdown-divider"></li>
-          <li class="dropdown-item <%= "active" if params[:sort_by] == "Newest" %>"><%= link_to "Newest", request.params.merge(sort_by: "Newest") %></li>
-          <li class="dropdown-item <%= "active" if params[:sort_by] == "Oldest" %>"><%= link_to "Oldest", request.params.merge(sort_by: "Oldest") %></li>
-          <li class="dropdown-item <%= "active" if params[:sort_by] == "Votes" %>"><%= link_to "Votes", request.params.merge(sort_by: "Votes") %></li>
+          <li class="dropdown-item <%= "active" if params[:order_column] == "created_at" && params[:order_direction] == "DESC" %>"><%= link_to "Newest", request.params.merge(order_column: "created_at", order_direction: "DESC") %></li>
```

Comment:
This code can be refactored like so:
In the Application helper for example:
```
def sortable_link(text, column, direction)
    active = params[:order_column] == column && params[:order_direction] == direction
    link_to(text, request.params.merge(order_column: column, order_direction: direction), 
    class: "dropdown-item #{'active' if active}")
end
```

In the view:

```
<ul class="dropdown-menu">
  <li class="dropdown-header">SORT BY</li>
  <li><hr class="dropdown-divider"></li>
  <li><%= sortable_link('Newest', 'created_at', 'DESC') %></li>
  <li><%= sortable_link('Oldest', 'created_at', 'ASC') %></li>
  <li><%= sortable_link('Votes', 'votes_count', 'DESC') %></li>
</ul>
```

WDYT?

Reply:
I will try this!